src/app/plugins/ariana.plugin.ts

```diff
@@ -0,0 +1,79 @@
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType, IHttpResponse, Maybe } from '../../common/types';
+
+export class ArianaPlugin extends Plugin {
+  public name: string = 'Ariana Plugin';
+  public description: string = 'Get a random picture of Ariana Grande.';
+  public usage: string = 'ariana';
+  public pluginAlias = ['ari'];
+  public permission: ChannelType = ChannelType.Public;
+
+  private _API_URL: string = 'https://www.gettyimages.com/photos/ariana-grande?page=';
+  private reImgs: RegExp = new RegExp(/<img class="gallery-asset__thumb gallery-mosaic-asset__thumb"(.*?)>/g);
+  private reThumb: RegExp = new RegExp(/https:\/\/(.*?)"/g);
+  private rePages: RegExp = new RegExp(/<span class="PaginationRow-module__lastPage___2pChH">(.*?)<\/span>/g);
+  private rePage: RegExp = new RegExp(/>(.*?)</g);
+  private nPages: number = 0;
+
+  constructor(public container: IContainer) {
+    super();
+
+    // get total number of pages of Ariana
+    this.container.httpService
+      .get(this._API_URL + '1')
+      .then((res: IHttpResponse) => {
+        const data: string = res.data;
+        const pagesSearch: Maybe<RegExpMatchArray> = data.match(this.rePages);
+
+        if (!pagesSearch) {
+          this.container.loggerService.warn('Failed to get Ariana page in constructor.');
+          return;
+        }
+        const pageSearch: Maybe<RegExpMatchArray> = pagesSearch[0].match(this.rePage);
+        if (pageSearch) {
+          this.nPages = parseInt(pageSearch[0].replace(/>|</g, ''));
+        }
+      })
+      .catch((err) => this.container.loggerService.warn(err));
+  }
+
+  private async getAri(page: number) {
+    let ari;
```

Comment:
im not really a fan of this, but if u feel strongly about it, u can keep it as is.

an alternative way (and more readble imo) could be;

```
const response = await this.container.httpService.get(this._API_URL + page).then(res => res.data);

// all logic within your `then` function can be moved outside since you can directly interact with `response`.
```
 this will get rid of the `ari` variable